client/state/login/selectors.js

```diff
@@ -175,6 +175,16 @@ export const getRedirectToSanitized = state => {
 	return get( state, 'login.redirectTo.sanitized', null );
 };
 
+/***
+ * Retrieves the last known value of `close_window_after_login` from the query params of a `/login` page.
+ *
+ * @param  {Object}   state  Global state tree
+ * @return {?bool}           True if the window should be closed after login.
+ */
+export const getCloseWindowAfterLogin = state => {
+	return get( state, 'login.closeWindowAfterLogin', null );
```

Comment:
Nit: arrays here used to be faster last time I looked.

```suggestion
	return get( state, [ 'login', 'closeWindowAfterLogin' ], null );
```